sheetfu/modules/table.py

```diff
@@ -39,34 +42,39 @@ def get_items_range(self):
             return None
         return self.full_range.offset(row_offset=1, column_offset=0, num_rows=(full_range_num_rows - 1))
 
-    def parse_items(self):
+    def parse_items(self, values):
         items = list()
         if not self.items_range:
             return items
+
+        notes = self.items_range.get_notes() if self.notes else None
+        backgrounds = self.items_range.get_notes() if self.backgrounds else None
```

Comment:
In this case I think you meant `self.items_range.get_backgrounds()`

In a perfect world `self.items_range.get_notes()`,  `self.items_range.get_backgrounds()`, etc should return a list of None values, but not sure if self.items_range has that info or not.

In case my previous suggestion is not possible, what about pre-initialising the lists to None so you avoid the ifs?
```python
empty_list = [None]*self.items_range.coordinates.number_of_rows

notes = self.items_range.get_notes() if self.notes else list(empty_list)
backgrounds = self.items_range.get_notes() if self.backgrounds else list(empty_list)
font_colors = self.items_range.get_notes() if self.font_colors else list(empty_list)

for row_number in range(0, self.items_range.coordinates.number_of_rows):
	item = Item(
		parent_table=self,
		row_index=row_number,
		header=self.header,
		values=values[row_number],
		notes=notes[row_number],
		backgrounds=backgrounds[row_number],
		font_colors=font_colors[row_number]
	)
```

Reply:
Completely true!